include/wsrep/transaction.hpp

```diff
@@ -142,6 +142,47 @@ namespace wsrep
         void xa_detach();
 
         int xa_replay(wsrep::unique_lock<wsrep::mutex>&);
+        int fragment_cache_remove_transaction(
+            const wsrep::id& server_id, wsrep::transaction_id transaction_id);
+        void *get_binlog_cache();
+        /* state of Streaming Replication Speedup feature for the
+           transaction. This describes the relationship of this WSREP
+           transaction and the underlying InnoDB transaction.
+        */
+        enum sr_state
```

Comment:
SR state handling could perhaps be moved to MariaDB side.

Reply:
Here we have the same question as above with `sr_store`. We need to  find a suitable place on the MariaDB side with a lifetime that is the same as the lifetime of the transaction.